# Software
[MACS2](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3868217/)
# Extract peak regions
## download human genome 
"All sequencing data was analyzed using the GRCh38/hg38 human reference genome."
- ensembl
[ensembl - current_fasta - homo_sapiens - dna](ftp://ftp.ensembl.org/pub/current_fasta/homo_sapiens/dna/)
ftp://ftp.ensembl.org/pub/current_fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.alt.fa.gz
strange chromosome name
- UCSC
[ucsc - hg38](http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/)
## download PeakCalls
[All cancer type-specific peak sets](https://api.gdc.cancer.gov/data/71ccfc55-b428-4a04-bb5a-227f7f3bf91c)
## extract fasta of peaks for a specific type of cancer and store them in a file
### test bedtools
[bedtools - getfasta](https://bedtools.readthedocs.io/en/latest/content/tools/getfasta.html#split-extracting-bed-blocks)
```bash
# test
# select first 3 fields in first 10 lines of the file
awk '{FS="\t";OFS="\t";{print $1,$2,$3;}}' Cancer_Type-specific_PeakCalls/ACC_peakCalls.bed | head > test.bed
# extract sequences after removing column names
bedtools getfasta -fi ../Homo_sapiens.GRCh38.dna.alt.fa -bed test.bed -fo test.bed.out
# check chromosome names
cat hg38.fa | grep ">" > chr_name_ucsc.txt
# select first 3 fields in first 10 lines of the file without column names
awk '{FS="\t";OFS="\t";if(FNR != 1){print $1,$2,$3;}}' ./PeakCalls/Cancer_Type-specific_PeakCalls/ACC_peakCalls.bed | head > test.bed
# extract sequences after removing column names
bedtools getfasta -fi ./hg38.fa -bed test.bed -fo test.bed.out
# remove column names and select the whole line in bed files to extract region 
awk '{FS="\t";OFS="\t";if(FNR != 1){print $0;}}' ACC_peakCalls.txt > ../../ACC_peakCalls.txt.bed
bedtools getfasta -fi ./hg38.fa -bed ACC_peakCalls.txt.bed -fo ACC_peakCalls.txt.bed.out


```
remove column names of .bed file
check chr names
### filter some peaks according to annotation
```
awk '{print $6}' Cancer_Type-specific_PeakCalls/ACC_peakCalls.txt | sort | uniq
```
3' /5' /Distal /Exon /Intron /Promoter
Exon should be removed? Where is PRDM9 binds to?
## liftOver

# Visualizing the bigwig files in R
## 5:00 - 6:19 download BigWig files
write BigWigLinks.txt and downloadLinks.sh to download files

<!--stackedit_data:
eyJoaXN0b3J5IjpbODg4Nzk3MzIzLC0zMTk0NTY3OTAsLTEwNz
A3NjAwNjEsLTYyNzIxNzc3OSwtNjU1OTMzNjg4LC04MTI5NjQ5
MzUsLTE1NDA5MTM5MDUsLTE3NDg1MjgwMzMsLTIxNzg1ODMxOC
wtNDY3MzAwMjYxLDE2MTI4MjYzNjcsMTExMjkyNzc3MiwtMTIz
MTA5NTkzMiwxODIxNTY4ODM3LC0xNzI1Mjc4MTgsLTU0Mjk0NT
E5OCwtMTU1NDY1ODYwOSw4NDIxMDQxNjgsLTcxOTA5ODIxMiw3
Mjg3NzgxMzhdfQ==
-->