# visualize BigWig
# ATAC-seq pipeline
# MACS BedGraph
# MACS peakcall and bdgpeakcall
```
wget "https://github.com/taoliu/MACS/raw/master/test/CTCF_SE_CTRL_chr22_50k.bed.gz"
wget "https://github.com/taoliu/MACS/raw/master/test/CTCF_SE_ChIP_chr22_50k.bed.gz"
```
## Step 1: Filter duplicates

Step 2: Decide the fragment length d
Step 3: Extend ChIP sample to get ChIP coverage track
Step 4: Build local bias track from control
The d background
The slocal background
The llocal background
The genome background
Combine and generate the maximum background noise
Step 5: Scale the ChIP and control to the same sequencing depth
Step 6: Compare ChIP and local lambda to get the scores in pvalue or qvalue
Step 7: Call peaks on score track using a cutoff
# bdgpeakcall parameters
<!--stackedit_data:
eyJoaXN0b3J5IjpbMTA5NTMzNjEzMiwtNjczNzI0NTA1LDk1Nz
YzMjU4XX0=
-->