# visualize BigWig
# ATAC-seq pipeline
# MACS BedGraph
# MACS peakcall and bdgpeakcall
```
wget "https://github.com/taoliu/MACS/raw/master/test/CTCF_SE_CTRL_chr22_50k.bed.gz"
wget "https://github.com/taoliu/MACS/raw/master/test/CTCF_SE_ChIP_chr22_50k.bed.gz"
```
write macs2_test.sh
## Step 1: Filter duplicates
By default, the maximum number of allowed duplicated reads is 1, or _--keep-dup=1_ for _callpeak_.
MACS2 will report the final number of reads kept after filtering
We need them when we have to scale the ChIP and control signals to the same depth. 
In this case, the number is 199583 for ChIP and 199867 for control, and the ratio between them is 199583/199867=.99858
Step 2: Decide the fragment length d
predict the length of the region of interest
Step 3: Extend ChIP sample to get ChIP coverage track
Step 4: Build local bias track from control
The d background
The slocal background
The llocal background
The genome background
Combine and generate the maximum background noise
Step 5: Scale the ChIP and control to the same sequencing depth
Step 6: Compare ChIP and local lambda to get the scores in pvalue or qvalue
Step 7: Call peaks on score track using a cutoff
# bdgpeakcall parameters
<!--stackedit_data:
eyJoaXN0b3J5IjpbMTc1NDUwMjk0MywxNjQ3NzI5NzA4LC02Mz
cwNTc1NzcsMTA5NTMzNjEzMiwtNjczNzI0NTA1LDk1NzYzMjU4
XX0=
-->