
```bash
# remove column names
awk '{FS=OFS="\t";if(FNR != 1){print $0;}}' ./PeakCalls/Cancer_Type-specific_PeakCalls/ACC_peakCalls.txt > ACC_peakCalls.bed
wc -l ./PeakCalls/Cancer_Type-specific_PeakCalls/ACC_peakCalls.txt # 90778
wc -l ACC_peakCalls.bed # 90777
bedtools getfasta -fi ./hg38.fa -bed ACC_peakCalls.bed -fo ACC_peakCalls.fasta
grep -c ">" ACC_peakCalls.fasta # 90777
bedtools getfasta -fi ./hg38.fa -bed ACC_peakCalls.bed -fo ACC_peakCalls_peakNames.fasta -name
grep -c ">" ACC_peakCalls_peakNames.fasta # 90777
awk 'BEGIN{FS=OFS="\t"}{if($0~/>/){name=$0; sub(">", "", $name); print $name}}' ACC_peakCalls_peakNames.fasta | wc -l # 90777
```
write extractSeq.sh to extract sequences of peaks in all types of cancer
use nohup.out to check number of sequences
use NCBI to check sequences: human chr1 501

store the PRDM9 13-basepair degenerate motif in meme format.
write fimo.sh to find PRDM9 motif in peaks from 23 types of cancer
```bash
# find PRDM9 motif in peaks from ACC
fimo --verbosity 4 -o ACC_peakCalls_fimo.out PRDM9.pwm.meme ../PeakFasta/ACC_peakCalls.fasta
# the number of peaks containing PRDM9 motif in ACC
awk '{FS=OFS="\t";print $1}' fimo.gff | sort | uniq | wc -l #32434
awk '{FS=OFS="\t";print $3}' fimo.tsv | sort | uniq | wc -l #32435


```
<!--stackedit_data:
eyJoaXN0b3J5IjpbLTE5ODE0NjY4NzUsLTIwOTgxODIxMTcsMT
k4NDI4NzQzNCwxOTE4NDgyNzY2LC0xNDkzMjQ1OTA5LDkyMDk0
MTkwMSwtMTQ1OTU2Mjc5MSwxODA1NzE4NTA4LDE2NzkyOTY4NT
IsMTM2MjY0MTQ1Nl19
-->