# BigWig Track Format
BigWig is used to store dense, continuous data that will be displayed in the Genome Browser as a graph.
# BedGraph Track Format
BedGraph is used to display continuous-valued data in track format. BedGraph can be transformed to BigWig to save time for displaying.
## general structure
```
chromA  chromStartA  chromEndA  dataValueA
chromB  chromStartB  chromEndB  dataValueB
```
> each line describes consecutive positions with the same coverage
> chromStartA  to chromEndA: consecutive positions. The intervals are sorted and non-overlapping.
> dataValue: coverage
> [About BedGraph](https://bedtools.readthedocs.io/en/latest/content/tools/genomecov.html#coverage-by-fragment)
> [Bed to BedGraph](https://toolshed.g2.bx.psu.edu/repository/display_tool?repository_id=1ec48b84b33d36d8&tool_config=database%2Fcommunity_files%2F000%2Frepo_121%2Fbedtools-galaxy%2FgenomeCoverageBed_bedgraph.xml&changeset_revision=41bba3e648d1)
# BigWig file  in Supplemental Data
normalized BigWig files for visualization of ATAC-seq tracks for each individual technical replicate
## source
[BigWig Files for All Samples](https://gdc.cancer.gov/about-data/publications/ATACseq-AWG)
## normalization method
The provided BigWig files have been normalized by the total insertions in peaks and then binned into 100-bp bins. 
Each 100-bp bin represents the normalized number of insertions that occurred within the corresponding 100 bp.
This is because samples of varying quality will have varying percentages of reads in peaks and cannot be adequately compared using depth normalization.
# Pipeling1
## convert BigWig to bedgraph
[BigWigtoBedGraph](https://genome.ucsc.edu/goldenPath/help/bigWig.html)
## call peaks from bedgraph
--shift -75 --extsize 150 --nomodel --call-summits --nolambda --keep-dup all -p 0.01
bdgpeakcall
[ATAC-seq analysis](https://yiweiniu.github.io/blog/2019/03/ATAC-seq-data-analysis-from-FASTQ-to-peaks/)
[call peaks from BedGraph](https://help.galaxyproject.org/t/macs-peaks-calling-from-bedgraph-data-chip-seq/1470)
[call peaks from BedGraph](https://cancerres.aacrjournals.org/content/canres/78/5/1127.full.pdf)
## extract peak regions for each techical replicates
## find PRDM9 motif within each peak region
Fimo
# Pipeline2
## convert BigWig to bedgraph
[WigtoBed](http://barcwiki.wi.mit.edu/wiki/SOPs/coordinates)
## call peaks from bedgraph
bdgpeakcall
## convert bedgraph to bam
[bedtobam](https://bedtools.readthedocs.io/en/latest/content/tools/bedtobam.html)
[BedGraph to Bam](https://www.jianshu.com/p/25f7a8967e8a)
### can not do it
- **I understand the information in BedGraph files, although I am still not sure why they used this method to normalize. As to the feasibility of pipeline2, I guess binding sites (the decrease in coverage within peak) are hard to distinguish because they are much more narrow than the 100bp-bin.** 
- [Don't convert Bigwig To Bam](https://www.biostars.org/p/49067/)
## footprint

# Other useful links
[WigtoBed](http://barcwiki.wi.mit.edu/wiki/SOPs/coordinates)
[explore ATAC-seq data](https://benbermanlab.com/assets/code/Workshop%20for%20ATAC-seq%20analysis.html)
[explore ATAC-seq data](https://www.google.com/search?q=TCGA+ATAC+counts+matrix+find+peaks+site:www.biostars.org&biw=1280&bih=578&sxsrf=ALeKk03sjIvoAhp6NkTxVgs0_uigKbzdLg:1590344874356&sa=X&ved=2ahUKEwj5oLmqkM3pAhViWhUIHXY2Ay8QrQIoBDACegQIARAP)
<!--stackedit_data:
eyJoaXN0b3J5IjpbMTg5NjA4NjU5NSw0OTgzMDQwMTUsMTk2Nj
I4NzAxMCw5MjE4OTM2MCwxMTQ0MzYyMjU0LC0xNzkxNzc4Mzky
LDEyNzM0Mjc0NTIsMzk1NDE3MDI0LDU4Nzc1ODA5OCw0ODk2MT
MzNSw4NzcxMTcwMTEsMTM4MjEyMTkzNSwtMTY5NzY0MjY1Mywt
MTE5MDI4Mjg5OSwxNDQ3MDE0NTEwLDEzMjUyODg1NzQsNjA5Mj
g0OTIwLDEwOTcxNjk4NjksLTExNTQ2NjIyNSwtMTI0MjU4NDA1
NF19
-->