# BigWig Track Format
BigWig is used to store dense, continuous data that will be displayed in the Genome Browser as a graph.
# BedGraph Track Format
BedGraph is used to display continuous-valued data in track format. BedGraph can be transformed to BigWig to save time for displaying.
## general structure
```
chromA  chromStartA  chromEndA  dataValueA
chromB  chromStartB  chromEndB  dataValueB
```
> each line describes consecutive positions with the same coverage
> chromStartA  to chromEndA: consecutive positions. The intervals are sorted and non-overlapping.
> dataValue: coverage
> [About BedGraph](https://bedtools.readthedocs.io/en/latest/content/tools/genomecov.html#coverage-by-fragment)
> [Bed to BedGraph](https://toolshed.g2.bx.psu.edu/repository/display_tool?repository_id=1ec48b84b33d36d8&tool_config=database%2Fcommunity_files%2F000%2Frepo_121%2Fbedtools-galaxy%2FgenomeCoverageBed_bedgraph.xml&changeset_revision=41bba3e648d1)
# BigWig file  in Supplemental Data
normalized BigWig files for visualization of ATAC-seq tracks for each individual technical replicate
## source
[BigWig Files for All Samples](https://gdc.cancer.gov/about-data/publications/ATACseq-AWG)
## normalization method
The provided BigWig files have been normalized by the total insertions in peaks and then binned into 100-bp bins. 
Each 100-bp bin represents the normalized number of insertions that occurred within the corresponding 100 bp.
This is because samples of varying quality will have varying percentages of reads in peaks and cannot be adequately compared using depth normalization.
# Pipeling1
## convert BigWig to bedgraph
[BigWigtoBedGraph](https://genome.ucsc.edu/goldenPath/help/bigWig.html)
## call peaks from bedgraph
bdgpeakcall
## extract peak regions for each techical replicates
## find PRDM9 motif with each peak region in 
Fimo
# Pipeline2
## convert BigWig to bedgraph
[WigtoBed](http://barcwiki.wi.mit.edu/wiki/SOPs/coordinates)
## call peaks from bedgraph
bdgpeakcall
## convert bedgraph to bam
I understand the information in BedGraph files, although I am still not sure why they used this method to normalize. For the feasibility of pipeline2, I guess because the binding sites are much more narrow than the 100bp-bin, the decrease in coverage within peak (binding sites) is hard to distinguish. 
## footprint

# Other useful links
[WigtoBed](http://barcwiki.wi.mit.edu/wiki/SOPs/coordinates)
<!--stackedit_data:
eyJoaXN0b3J5IjpbODIxNjUyNzgsNDg5NjEzMzUsODc3MTE3MD
ExLDEzODIxMjE5MzUsLTE2OTc2NDI2NTMsLTExOTAyODI4OTks
MTQ0NzAxNDUxMCwxMzI1Mjg4NTc0LDYwOTI4NDkyMCwxMDk3MT
Y5ODY5LC0xMTU0NjYyMjUsLTEyNDI1ODQwNTQsMTc0NTI3MjU2
Myw0OTYyNTQ3MjEsMTc4NDQ0MzMzNSw1MzEwNjYxNDYsMjExMD
YxMzMxMCwyMDg2NDU1NzU1XX0=
-->