# bigWig Track Format
## definition
bigWig is used to store dense, continuous data that will be displayed in the Genome Browser as a graph.
# BedGraph Track Format
## definition
BedGraph is used to display continuous-valued data in track format. BedGraph can be transformed to bigWig to save time for displaying.
## General Structure
```
chromA  chromStartA  chromEndA  dataValueA
chromB  chromStartB  chromEndB  dataValueB
```

	> each line describes consecutive positions with the same coverage
	> chromStartA  to chromEndA: consecutive positions
	> dataValue: coverage
	> [About BedGraph](https://bedtools.readthedocs.io/en/latest/content/tools/genomecov.html#coverage-by-fragment)
	> [Bed to BedGraph](https://toolshed.g2.bx.psu.edu/repository/display_tool?repository_id=1ec48b84b33d36d8&tool_config=database%2Fcommunity_files%2F000%2Frepo_121%2Fbedtools-galaxy%2FgenomeCoverageBed_bedgraph.xml&changeset_revision=41bba3e648d1)
# BigWig file in this paper
normalized bigWig files for visualization of ATAC-seq tracks for each individual technical replicate
The provided bigWig files have been normalized by the total insertions in peaks and then binned into 100-bp bins. Each 100-bp bin represents the normalized number of insertions that occurred within the corresponding 100 bp.
# Pipeling1
## convert BigWig to bedgraph
[BigWigtoBedGraph](https://genome.ucsc.edu/goldenPath/help/bigWig.html)
## call peaks from bedgraph
bdgpeakcall
## extract peak regions for each techical replicates
## find PRDM9 motif with each peak region in 
Fimo
# Pipeline2
## convert BigWig to bedgraph
[WigtoBed](http://barcwiki.wi.mit.edu/wiki/SOPs/coordinates)
## call peaks from bedgraph
bdgpeakcall
## convert bedgraph to bam
## footprint

# Other useful links
[WigtoBed](http://barcwiki.wi.mit.edu/wiki/SOPs/coordinates)
<!--stackedit_data:
eyJoaXN0b3J5IjpbLTE5NjMzNjI0OSwxNDQ3MDE0NTEwLDEzMj
UyODg1NzQsNjA5Mjg0OTIwLDEwOTcxNjk4NjksLTExNTQ2NjIy
NSwtMTI0MjU4NDA1NCwxNzQ1MjcyNTYzLDQ5NjI1NDcyMSwxNz
g0NDQzMzM1LDUzMTA2NjE0NiwyMTEwNjEzMzEwLDIwODY0NTU3
NTVdfQ==
-->